### PR TITLE
zephyr: renamed single-image mode to single-slot mode

### LINKED ISSUE #800: Single-image boot is ambiguous
The terms "single-image boot" and the "single image application" of Zephyr are confused.
The original suggestion comes from @nvlsianpu (https://github.com/JuulLabs-OSS/mcuboot/pull/793#discussion_r487111576).
It is not clear whether the MCUBoot handles only one updateable image with the "classic" primary + secondary image slot model or there is only 1 image slot in total.
Would it make sense to have this single slot/image mode in case of multiple independent images?
The documentation and comments should be rephrased to clearly distinguish these 2 cases.

### PR DESCRIPTION
**Pull request by nvlsianpu**
The terms "single-image boot" and the "single image application"
of Zephyr are confused. It might be might be understood as one-pair image
dual-bank mode in comparison to multi-image mode.

This patch replaces CONFIG_SINGLE_IMAGE_DFU by CONFIG_SINGLE_SLOT_DFU
name, which makes this mode name unambiguous.

fixes #800

zephyr CI: https://github.com/zephyrproject-rtos/zephyr/pull/28396

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
@de-nordic ?

**Comment by de-nordic** [comment]
I do not mind changing the name. Unfortunately I had not gave enough though to the name prior to crafting that motley of words.
Generally it should mean "The one application slot configuration" so maybe it should be rather replaced with
`SINGLE_APPLICATION_SLOT`. I think that the `DFU` is misleading here because the code does not actually provide DFU functionality, it is provided by other components of the MCUBOOT; even the source file name indicates that this is only loader (`single_loader.c`).

**Comment by nvlsianpu** [comment]
@de-nordic `SINGLE_APPLICATION_SLOT` sound good to me as well. And, right DFU abbreviation  has nothing to what MCUBoot does.

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -58,7 +58,7 @@ config NRF_CC310_BL
 
 menu "MCUBoot settings"
 
-config SINGLE_IMAGE_DFU
+config SINGLE_APPLICATION_SLOT
 	bool "Single image application"
```
Comment (de-nordic) [inline boot/zephyr/Kconfig:62]: Maybe change the title here to be also more in line with the function the config enables.

**Review by de-nordic — APPROVED** [review]
Looks fine, compiles and works as tested on nrf52840dk.

Approved with come comments left (https://github.com/JuulLabs-OSS/mcuboot/pull/801#discussion_r488593316)

**Comment by de-nordic** [comment]
@davidvincze Do you want to look at this?

File: boot/zephyr/Kconfig (hunk 1 above)
Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: Is it better now?

Comment (de-nordic) [reply boot/zephyr/Kconfig]: .Kinda sounds like there is a feature where application may span multiple slots.

Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: Actually this was same before.

Comment (nvlsianpu) [reply boot/zephyr/Kconfig]: `single slot bootloader`?

Comment (de-nordic) [reply boot/zephyr/Kconfig]: I think that this sounds better.

**Review by davidvincze — APPROVED** [review]
LGTM

**Pull request merged**